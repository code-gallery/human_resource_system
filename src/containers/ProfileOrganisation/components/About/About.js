import React from 'react'
import PropTypes from 'prop-types'
import ContentCard from 'components/ContentCard'

const About = (props) => {
  const { about_us } = props.organisation
  const aboutHTML = (about_us) ? about_us.replace(/[\r\n][\n\n]/g, '<br />') : false

  return (
    <ContentCard title="About">
      {aboutHTML &&
        <p dangerouslySetInnerHTML={{ __html: aboutHTML }}></p>
      }
    </ContentCard>
  )
}

About.propTypes = {
  organisation: PropTypes.shape({
    about_us: PropTypes.string
  })
}

export default About
