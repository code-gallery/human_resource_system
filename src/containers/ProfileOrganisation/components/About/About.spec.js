import React from 'react'
import renderer from 'react-test-renderer'
import About from './index'

describe('containers/ProfileOrganisation/components/About', () => {
  describe('when about_us is null', () => {
    it('renders correctly', () => {
      const props = {
        organisation: {
          about_us: null
        }
      }
      const tree = renderer.create(
        <About {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })

  describe('when about_us is valid', () => {
    it('renders correctly', () => {
      const props = {
        organisation: {
          about_us: 'Hello world\n\nLorem ipsum dolor\r\n sit amet\n blah blah'
        }
      }
      const tree = renderer.create(
        <About {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })
})
