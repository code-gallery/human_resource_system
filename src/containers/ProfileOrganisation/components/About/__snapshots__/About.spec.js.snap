// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/ProfileOrganisation/components/About when about_us is null renders correctly 1`] = `
<div
  className="ContentCard col-xs-12 "
>
  <div
    className="ContentCard-wrapper"
  >
    <h3
      className="ContentCard-title"
    >
      About
    </h3>
  </div>
</div>
`;

exports[`containers/ProfileOrganisation/components/About when about_us is valid renders correctly 1`] = `
<div
  className="ContentCard col-xs-12 "
>
  <div
    className="ContentCard-wrapper"
  >
    <h3
      className="ContentCard-title"
    >
      About
    </h3>
    <p
      dangerouslySetInnerHTML={
        Object {
          "__html": "Hello world<br />Lorem ipsum dolor<br /> sit amet
         blah blah",
        }
      }
    />
  </div>
</div>
`;
